packages/client/src/generation/dmmf.ts

```diff
@@ -108,96 +109,104 @@ class DMMFSchemaHelper implements Pick<DMMF.Document, 'schema'> {
     }
   }
 
-  resolveOutputTypes() {
-    for (const type of this.outputTypes.model) {
+  private resolveOutputTypes() {
+    this.resolveOutputTypesInNamespace('prisma')
+    this.resolveOutputTypesInNamespace('model')
+  }
+
+  private resolveOutputTypesInNamespace(namespace: DMMF.FieldNamespace) {
+    for (const type of this.outputTypes[namespace]) {
       for (const field of type.fields) {
-        if (typeof field.outputType.type === 'string' && !ScalarTypeTable[field.outputType.type]) {
-          field.outputType.type =
-            this.outputTypeMap[field.outputType.type] ||
-            this.outputTypeMap[field.outputType.type] ||
-            this.enumMap[field.outputType.type] ||
-            field.outputType.type
+        if (typeof field.outputType.type !== 'string') {
+          // should not happen
+          continue
         }
-      }
-      type.fieldMap = keyBy(type.fields, 'name')
-    }
 
-    for (const type of this.outputTypes.prisma) {
-      for (const field of type.fields) {
-        if (typeof field.outputType.type === 'string' && !ScalarTypeTable[field.outputType.type]) {
```

Comment:
Was the scalar table not needed anymore because you changed the way the condition is checked?